src/CAN/can_mapping.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Race Capture Firmware
+ *
+ * Copyright (C) 2016 Autosport Labs
+ *
+ * This file is part of the Race Capture firmware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "can_mapping.h"
+#include "byteswap.h"
+#include <byteswap.h>
+
+#include <stdio.h>
+
+float canmapping_extract_value(uint64_t raw_data, CANMapping *mapping)
+{
+	uint8_t offset = mapping->offset;
+	uint8_t length = mapping->length;
+	if (! mapping->bit_mode) {
```

Comment:
Customers need to express it both ways, but most of the time they're thinking in whole byte offset and lengths, and it's not reasonable to force them to think in bits. So, we need the flexibility to express it both ways. Happy to chat voice about it next opportunity.